### DISCUSSION TITLE
CSS Article filter not working with img

### DISCUSSION THREAD

**Discussion by sherlcok314159** [FreshRSS/FreshRSS#6217] [Q&A] [answered]
**Describe the bug**

CSS article filter not working with img

**To Reproduce**

You can try this [feed](https://raw.githubusercontent.com/sherlcok314159/Files_/main/%E6%9E%81%E5%B8%82%E5%B9%B3%E5%8F%B0.xml) with a full-article selector as `#js_content`. And when you select the first img in the article to filter. It is not working.

<img width="1121" alt="图片" src="https://github.com/FreshRSS/FreshRSS/assets/76043326/f23d82ad-a871-4be9-a1b8-a8adf1052623">

The first filter works fine.

```
mp-style-type, img[src="https://mmbiz.qpic.cn/sz_mmbiz_gif/gYUsOT36vfoGnnrz1tAAA0UNPxQaxTGM7WpEmJcx4UPcaibm3FUZpY6Jzicic0sic2fzYicoogTIzwnbNicicos17UGibw/640?wx_fmt=gif&wxfrom=5&wx_lazy=1&wx_co=1"]
```

 #And the CSS selector should be correct.

<img width="1624" alt="图片" src="https://github.com/FreshRSS/FreshRSS/assets/76043326/0a7314e4-98ef-42c8-bc07-517e586207e3">

**Environment information (please complete the following information):**

    Device: MacOS
    OS: Ubuntu 20.04
    Browser: Firefox / Edge
    FreshRSS version: 1.23.1
    PHP version: 8.2.7
    Installation type: Docker-compose

#4501

**Comment by Frenzie** [comment, marked as answer]
The attribute is `data-src`. The JS on that page uses that to turn it into `src`. (You're lucky in fact that despite being contained in something named `#js_content` it's actually there in plain HTML. ;-)

**Comment by sherlcok314159** [comment]
Thanks for your help. It works like a charm! I want to ask if are there any other similar examples (e.g., some attribute needs to be transformed into another). I am new to this stuff :-).

**Reply by Frenzie** [reply]
If you use right click, view source on the page you'll normally see exactly what FreshRSS sees. Anything that's loaded in later by JS is not supported. You can also use the inspector on it if you disable JS or use NoScript.

**Reply by math-GH** [reply]
Also useful: The source view

![grafik](https://github.com/FreshRSS/FreshRSS/assets/1645099/792671ef-bb9d-4c9b-b873-35a9299f17bc)
![grafik](https://github.com/FreshRSS/FreshRSS/assets/1645099/d3685c9a-41cb-4b7d-8562-43bcd2e15fa7)

**Comment by sherlcok314159** [comment]
WOW. Much thanks. Now I know the stuff behind the filter.
